src/styles/bannerCommon.ts

```diff
@@ -238,3 +238,261 @@ export const styles = {
         }
     `,
 };
+
+interface StyleData {
+    styleBackground: string;
+    styleHeader: string;
+    styleBody: string;
+    styleHighlight: string;
+    styleHighlightBackground: string;
+    styleButton: string;
+    styleButtonBackground: string;
+    styleButtonHover: string;
+    styleCloseButton: string;
+}
+
+export const selfServeStyles = (style: StyleData) => {
+    return {
+        wrapper: css`
+            box-sizing: border-box;
+            display: flex;
+            flex-direction: column;
+            justify-content: center;
+            align-items: center;
+            width: 100%;
+            background-color: ${style.styleBackground};
```

Comment:
I wonder if we should be sanitising this user input before interpolating into the styles? I'm not worried about malicious intent but more accidentally pasting bad content in which could break the page. What do you think? It's possible to add further styles here by adding `;` to the prop/value passed from Braze, for example changing `styleBackground` to `red; transform: rotate(90deg)` gives: 
![Screenshot 2023-06-30 at 15 03 44](https://github.com/guardian/braze-components/assets/379839/f25fbca6-9a47-454d-85bc-48065c3670c6)

Perhaps we could add some simple validation to the `canRender` method to ensure only values which match `rbd(\d?, \d?, \d?)` or similar? We could even parse the supplied string into something with the following type, if validation passes:

```
interface ColourStyle {
  r: number,
  g: number,
  b: number
}
```

This would allow us to give richer type information in the function signature of `selfServeStyles` (`ColourStyle` tells the developer more than `string`) and would make it impossible to pass arbitrary strings in to be interpolated into the styles.

Reply:
That's true, there are many different formats, but I don't think we need to support them here. I think it would be a worthwhile trade-off to limit the flexibility to one (or two) formats because of the extra safety it would give and validation it would allow us to do.

Reply:
A fix to prevent malicious/accidental css has been pushed up - also fixes the default CSS colors issue.

Reply:
To fix the issue of malformed color strings I've pushed up an additional fix which works by applying default colors in the CSS immediately before the required color (which may not conform to CSS color definitions). If the required color is a malformed color string then the CSS engine will skip it and the displayed colour will fallback to the default color